src/audio/host-zephyr.c

```diff
@@ -836,8 +837,10 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
 	}
 
 	/* calculate DMA buffer size */
-	buffer_size = ALIGN_UP(period_bytes, align) * period_count;
-	buffer_size = MAX(buffer_size, ALIGN_UP(hd->ipc_host.dma_buffer_size, align));
+	round_up_size = (params->frame_fmt == SOF_IPC_FRAME_S24_3LE) ? (3 * align) : align;
+	buffer_size = ROUND_UP(period_bytes, round_up_size) * period_count;
+	if (hd->ipc_host.dma_buffer_size != 0)
+		buffer_size = ROUND_UP(hd->ipc_host.dma_buffer_size, buffer_size);
```

Comment:
@fkwasowi it still looks strange to me - why we potentially almost double the buffer size. Say, if `buffer_size` is 4096 bytes and `dma_buffer_size` is just a bit larger, like 4100, this will set the buffer size to double the original size - to 8192 bytes, is this what is needed?